impl/src/main/java/org/ehcache/internal/store/heap/OnHeapStore.java

```diff
@@ -1027,7 +1028,13 @@ private void checkValue(V valueObject) {
       throw new ClassCastException("Invalid value type, expected : " + valueType.getName() + " but was : " + valueObject.getClass().getName());
     }
   }
-  
+
+  private void onExpiration(K mappedKey, ValueHolder<V> mappedValue) {
+    expirationObserver.begin();
+    expirationObserver.end(StoreOperationOutcomes.ExpirationOutcome.SUCCESS);
```

Comment:
Not a fan of firing the observers like this, but there's probably nothing else we can do here.